api/Trace/SpanStatus.php

```diff
@@ -6,45 +6,45 @@
 
 interface SpanStatus
 {
-    const OK = 0;
-    const CANCELLED = 1;
-    const UNKNOWN = 2;
-    const INVALID_ARGUMENT = 3;
-    const DEADLINE_EXCEEDED = 4;
-    const NOT_FOUND = 5;
-    const ALREADY_EXISTS = 6;
-    const PERMISSION_DENIED = 7;
-    const RESOURCE_EXHAUSTED = 8;
-    const FAILED_PRECONDITION = 9;
-    const ABORTED = 10;
-    const OUT_OF_RANGE = 11;
-    const UNIMPLEMENTED = 12;
-    const INTERNAL = 13;
-    const UNAVAILABLE = 14;
-    const DATA_LOSS = 15;
-    const UNAUTHENTICATED = 16;
```

Comment:
If we're concerned about memory footprint, we could always keep these constant numbers and add another array similar to `DESCRIPTION []` but for the status title?  I don't know what the real world implication of changing this from in to string at scale is going to be.

Reply:
I want to make a memory footprint test to measure the impact.

On the other hand, the specs say that one should be able to pass its own Code and Description and that this list is not exhaustive. If we decide to keep the integer codes then we have to allow 2 types of code, int for current defined codes and strings for custom codes.

[StatusCanonicalCodes](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/trace/api.md#statuscanonicalcode) points to [Standard GRPC Codes](https://github.com/grpc/grpc/blob/master/doc/statuscodes.md) which says: 

> The gRPC client and server-side implementations may also generate and return status on their own when errors happen. Only a subset of the pre-defined status codes are generated by the gRPC libraries. This allows applications to be sure that any other code it sees was actually returned by the application (although it is also possible for the server-side to return one of the codes generated by the gRPC libraries).

Reply:
I just finished my tests and there is no impact on memory footprint.

Test: 
```php
for ($i = 0; $i < 10000000; $i++) {
    $span = $tracer->startAndActivateSpan('span' . $i);
    $status = $statusCodes[array_rand($statusCodes)];

    $span->setSpanStatus(
        constant("OpenTelemetry\Trace\SpanStatus::$status"),
        SpanStatus::DESCRIPTION[constant("OpenTelemetry\Trace\SpanStatus::$status")]
    );

    if ($i % 1000000 == 0) {
        echo PHP_EOL . "Memory usage: ". (memory_get_usage() / 1024 / 1024) . " MB" ;
    }
}
```

status stored as integers:
```
Memory usage: 1.6644973754883 MB
Memory usage: 869.70763397217 MB
Memory usage: 1737.7589035034 MB
Memory usage: 2605.8101730347 MB
Memory usage: 3473.8614425659 MB
```
status stored as string:
```
Memory usage: 1.6651077270508 MB
Memory usage: 869.70824432373 MB
Memory usage: 1737.759513855 MB
Memory usage: 2605.8107833862 MB
Memory usage: 3473.8620529175 MB
```

I think we can stores statuses as strings.